src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -786,6 +786,46 @@ internal HttpRequestMessage BuildSetTriggersRequest()
             }
         }
 
+        public async Task<TriggersResult> GetTriggersAsync()
+        {
+            var result = new TriggersResult
+            {
+                Success = true
+            };
+
+            if (!IsSyncTriggersEnvironment(_webHostEnvironment, _environment))
+            {
+                result.Success = false;
+                result.Error = "Invalid environment for GetTriggers operation.";
+                _logger.LogWarning(result.Error);
+                return result;
+            }
+
+            try
+            {
+                await _syncSemaphore.WaitAsync();
```

Comment:
I was going to point out that if we're going to be synchronizing here as well, we'd rename the semaphore to _triggerOperationSemaphore or something. However, I don't think we even need syncronization here and can just remove it from this method. The synchronization is needed for SyncTriggers because it's performing multiple steps that need to be in sync. For the get API we don't need this.

Reply:
makes sense, removed it.